annotation-server/src/medications/medications.service.ts

```diff
@@ -29,66 +31,67 @@ export class MedicationsService {
         private medicationRepository: Repository<Medication>,
     ) {}
 
-    getAll(
-        options: FindManyOptions<Medication> = {
-            select: ['id', 'name', 'description', 'synonyms', 'drugclass'],
-        },
+    async findAll(
+        limit: number,
+        offset: number,
+        search: string,
+        sortBy: string,
+        orderBy: string,
+        withGuidelines: boolean,
+        onlyIds: boolean,
     ): Promise<Medication[]> {
-        return this.medicationRepository.find(options);
-    }
-
-    getWithGuidelines(): Promise<Medication[]> {
-        return this.medicationRepository.find({
-            where: {
-                guidelines: {
-                    id: Not(IsNull()),
-                },
+        if (onlyIds) return this.getAllIds();
+
+        const whereClause: FindOptionsWhere<Medication> = {};
+        const findOptions = <FindManyOptions<Medication>>{
+            where: whereClause,
+            take: limit,
+            skip: offset,
+            order: {
+                [sortBy]: orderBy === 'asc' ? 'ASC' : 'DESC',
             },
-            relations: [
+        };
+
+        if (search) {
+            const matchingIds = await this.findIdsMatching(search);
+            whereClause.id = In(matchingIds);
+        }
+
+        if (withGuidelines) {
+            whereClause.guidelines = { id: Not(IsNull()) };
+            findOptions.relations = [
                 'guidelines',
                 'guidelines.genePhenotype.phenotype',
                 'guidelines.genePhenotype.geneSymbol',
-            ],
-        });
+            ];
+        }
+
+        return await this.medicationRepository.find(findOptions);
     }
 
-    getOne(options: FindOneOptions<Medication>): Promise<Medication> {
-        return this.medicationRepository.findOneOrFail(options);
+    async getAllIds(): Promise<Medication[]> {
```

Comment:
Why not `findAllIds`?

Reply:
Wanted to create contrast between classic CRUD methods that are relevant to the controller and 'helper methods'